test/testautomation_video.c

```diff
@@ -821,28 +837,57 @@ static int video_getSetWindowPosition(void *arg)
 {
     const char *title = "video_getSetWindowPosition Test Window";
     SDL_Window *window;
+    int maxxVariation, maxyVariation;
     int xVariation, yVariation;
     int referenceX, referenceY;
     int currentX, currentY;
     int desiredX, desiredY;
+    SDL_Rect display_bounds;
 
     /* Call against new test window */
     window = createVideoSuiteTestWindow(title);
     if (!window) {
         return TEST_ABORTED;
     }
 
-    for (xVariation = 0; xVariation < 4; xVariation++) {
-        for (yVariation = 0; yVariation < 4; yVariation++) {
+    /* Sanity check */
+    SDL_GetWindowPosition(window, &currentX, &currentY);
+    if (SDL_SetWindowPosition(window, currentX, currentY) < 0) {
+        SDLTest_Log("Skipping window positioning tests: %s reports window positioning as unsupported", SDL_GetCurrentVideoDriver());
+        goto null_tests;
+    }
+
+    if (SDL_strcmp(SDL_GetCurrentVideoDriver(), "x11") == 0) {
+        /* Platform forces windows to be within desktop bounds */
```

Comment:
Only if there's a window manager. Some kiosk type use-cases have a bare X server or minimal window manager that doesn't have any desktop bounds restrictions.

Reply:
Hmm, I'm wondering how to handle this. Check the `XDG_CURRENT_DESKTOP` envvar and only impose the restriction if a window manager is present? Just check for GNOME and KDE, which I know have this restriction, and for everything else, let the chips fall where they may?

Reply:
I think it's fine to ignore this, or maybe just document it. I just wanted to make sure you weren't assuming that was true for all environments in the backend code.

Reply:
Yes, no assumptions about this are made in the backend itself, it's just needed in this test so it passes on GNOME and KDE. It's now documented as such.